src/webgpu/shader/execution/builtin/asin.spec.ts

```diff
@@ -0,0 +1,66 @@
+export const description = `
+Execution Tests for the 'asin' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { f32, TypeF32 } from '../../../util/conversion.js';
+
+import { ulpThreshold, Case, Config, run } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('float_builtin_functions,asin')
+  .uniqueId('322c7c5ba84c257a')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#float-builtin-functions')
+  .desc(
+    `
+asin:
+T is f32 or vecN<f32> asin(e: T ) -> T Returns the arc sine of e. Component-wise when T is a vector. (GLSLstd450Asin)
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    // TODO(https://github.com/gpuweb/cts/issues/792): Decide what the ground-truth is for these tests.
+    const truth_func = (x: number): number => {
+      return Math.asin(x);
+    };
+
+    // Well defined/border cases
+    const manual_inputs: Array<number> = [
```

Comment:
For manual, well known values like this, I think it would be nicer to just explicitly declare the expected value (can be done using multiples / fractions of `PI`). Especially when you're relying on the JS engine's implementation around trig and DBZs / `NaN`s.